packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -997,3 +959,1336 @@ class _RenderTheatre extends RenderBox with ContainerRenderObjectMixin<RenderBox
+    return now;
+  }
+
+  /// Show the overlay child of the [OverlayPortal] this controller is attached
+  /// to, at the top of the target [Overlay].
+  ///
+  /// When there are more than one [OverlayPortal]s that target the same
+  /// [Overlay], the overlay child of the last [OverlayPortal] to have called
+  /// [show] appears at the top level, unobstructed.
+  ///
+  /// If [isShowing] is already true, calling this method brings the overlay
+  /// child it controls to the top.
+  ///
+  /// This method should typically not be called when the widget tree is being
+  /// rebuilt.
+  void show() {
+    final _OverlayPortalState? state = _attachTarget;
+    if (state != null) {
+      state.show(_now());
+    } else {
+      _showTimestamp = _now();
+    }
+  }
+
+  /// Hide the [OverlayPortal]'s overlay child.
+  ///
+  /// Once hidden, the overlay child will be removed from the widget tree the
+  /// next time the widget tree rebuilds, and stateful widgets in the overlay
+  /// child may lose states as a result.
+  ///
+  /// This method should typically not be called when the widget tree is being
+  /// rebuilt.
+  void hide() {
+    final _OverlayPortalState? state = _attachTarget;
+    if (state != null) {
+      state.hide();
+    } else {
+      assert(_showTimestamp != null);
+      _showTimestamp = null;
+    }
+  }
+
+  /// Whether the associated [OverlayPortal] should build and show its overlay
+  /// child, using its `overlayChildBuilder`.
+  bool get isShowing {
+    final _OverlayPortalState? state = _attachTarget;
+    return state != null
+      ? state._showTimestamp != null
+      : _showTimestamp != null;
+  }
+
+  /// Conventience method for toggling the current [isShowing] status.
+  ///
+  /// This method should typically not be called when the widget tree is being
+  /// rebuilt.
+  void toggle() => isShowing ? hide() : show();
+
+  @override
+  String toString() {
+    final String? debugLabel = _debugLabel;
+    final String label = debugLabel == null ? '' : '($debugLabel)';
+    final String isDetached = _attachTarget != null ? '' : ' DETACHED';
+    return '${objectRuntimeType(this, 'OverlayPortalController')}$label$isDetached';
+  }
+}
+
+/// A widget that renders its overlay child on the closest [Overlay].
+///
+/// The overlay child is initially hidden until [OverlayPortalController.show]
+/// is called on the associated [controller]. The [OverlayPortal] uses
+/// [overlayChildBuilder] to build its overlay child and renders it on the
+/// closest [Overlay] as if it was inserted using an [OverlayEntry], while it
+/// can depend on the same set of [InheritedWidget]s (such as [Theme]) that this
+/// widget can depend on.
+///
+/// This widget requires an [Overlay] ancestor in the widget tree when its
+/// overlay child is showing. The overlay child will be painted in front of the
+/// [OverlayEntry] closest to this widget in the widget tree (which is usually
```

Comment:
What defines closeness for OverlayEntries? I can see what the closest Overlay is, but what's the closest OverlayEntry in that Overlay?